### PR TITLE
[Az.Sql] Add support for cross-subscription Azure SQL Database Failover Group creation

### PR DESCRIPTION
**Pull request by SharanSingh**
## Description

This PR adds a new parameter in New-AzSqlDatabaseFailoverGroup cmdlet to support passing in a different subscription Id for partner side of an Azure Sql Database Failover Group. Cross-subscription failover group creation is already supported via ARM templates for more than a year. This PR brings that functionality to Azure Powershell.

Approved Issue: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1230

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from the scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** change version of module in pull request

### CODE REVIEW

File: src/Sql/Sql.Test/ScenarioTests/FailoverGroupTests.cs [hunk 1]
```diff
@@ -84,6 +84,13 @@ public void TestCreateFailoverGroup_Overflow()
             RunPowerShellTest("Test-CreateFailoverGroup-Overflow");
         }
 
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestCreateFailoverGroup_CrossSubscription()
+        {
+            RunPowerShellTest("Test-CreateFailoverGroup-CrossSubscription");
```
Comment (BethanyZhou) [inline src/Sql/Sql.Test/ScenarioTests/FailoverGroupTests.cs:91]: As our test framework is adjusted, please pull latest main into your branch and change this Line to 
```suggestion
            TestRunner.RunTestScript("Test-CreateFailoverGroup-CrossSubscription");
```

File: src/Sql/Sql/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,8 @@
 -->
 ## Upcoming Release
```
Comment (BethanyZhou) [inline src/Sql/Sql/ChangeLog.md:21]:
```suggestion
```

File: src/Sql/Sql/help/New-AzSqlDatabaseFailoverGroup.md [hunk 3]
```diff
@@ -41,6 +42,13 @@ $failoverGroup = New-AzSqlDatabaseFailoverGroup -ResourceGroupName rg1 -ServerNa
 
 This command creates a new Failover Group with failover policy 'Manual' for two servers in different resource groups.
 
+### Example 3
+```powershell
+$failoverGroup = New-AzSqlDatabaseFailoverGroup -ResourceGroupName rg1 -ServerName primaryserver -PartnerSubscriptionId subscription2 -PartnerServerName secondaryserver1 -FailoverGroupName fg -FailoverPolicy Manual
+```
```
Comment (BethanyZhou) [inline src/Sql/Sql/help/New-AzSqlDatabaseFailoverGroup.md:48]: Please add sample output with following pattern if possible 
\```output
{output}
\```

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Sql/Sql/ChangeLog.md (hunk 2 above)
Comment (SharanSingh) [reply src/Sql/Sql/ChangeLog.md]: Removed extra line

File: src/Sql/Sql.Test/ScenarioTests/FailoverGroupTests.cs (hunk 1 above)
Comment (SharanSingh) [reply src/Sql/Sql.Test/ScenarioTests/FailoverGroupTests.cs]: Pulled in latest changes from Azure/azure-powershell:main into SharanSingh/azure-powershell:main and updated the test method name from `RunPowerShellTest` to `TestRunner.RunTestScript`

File: src/Sql/Sql/help/New-AzSqlDatabaseFailoverGroup.md (hunk 3 above)
Comment (SharanSingh) [reply src/Sql/Sql/help/New-AzSqlDatabaseFailoverGroup.md:64]: Added sample output.

**Review by BethanyZhou — APPROVED** [review]
Looks good to me

**Comment by SharanSingh** [comment]
@BethanyZhou how do I rerun an Azure pipeline job on this PR. One of the jobs seems to have been cancelled (reported as failed) and it is blocking check-in of this PR.

**Pull request merged**